pkg/collector/corechecks/system/iostats_nix.go

```diff
@@ -50,11 +50,17 @@ func roundFloat(val float64) float64 {
 
 // Can't use values from math.MaxUint* because C size for longs changes on 32/64 bit machines
 const maxLong = int64(^uint(0) >> 1)
```

Comment:
Not directly related to your changes, but I don't understand why we set this value to the maximum `long` value instead of the maximum `unsigned long` value. I think the unit tests pass only because we re-use that value there.

In any case, I agree with @albertvaka here, if we go with the approach of guessing what the max values are, this constant isn't needed anymore.

Reply:
This came from the assumption that some values wrapped because they were still 64 bits but signed. But it's not the case.

Reply:
Consider the call `incrementWithOverflow(0, math.MaxInt64)`
the value of `ret := int64(currentValue - lastValue)` will be `math.MinInt64+1`
You want to do `ret = ret + math.MaxInt64` to get the expect value `1` (math.MinInt64=-9223372036854775808 and math.MaxInt64=9223372036854775807)

Adding UlongMax will result in decrementing the value
```
var v int32 = 42
var ulong32Max uint32 = math.MaxUint32
// cast to int64 is to bypass overflow check
fmt.Println(int32(int64(v) + int64(ulong32Max))) // Print 41 
```